lib/js/jslib.js

```diff
@@ -1757,11 +1757,30 @@ function _getInputValue(id) {
   var element = document.getElementById(id);
   DEBUG.assert(element != null, "invalid input node (id " + id + ")");
   DEBUG.assert(element.value != undefined, "invalid input value in id " + id)
-  return element.value;
+  if ((element.type != "radio" && element.type != "checkbox") || element.checked) {
```

Comment:
If `element.type` is always (supposed to be) a string, then I think you want to use strict inequality `!==` in both instances.

Reply:
Hmm.  it might be undefined (if the element whose id we looked up lacked the `type` attribute).  If the element type is undefined then I think we should just return the value (this is most likely due to a bug in the form, and we are not trying very hard to check for this class of bugs).  We could be more careful here by checking that each input field in a form does have a type in `desugarLAttributes`.

Reply:
> We could be more careful here by checking that each input field in a form does have a type in `desugarLAttributes`.

That sounds entirely sensible to me!

Reply:
While that would be a good idea, it would be stricter than (de facto) HTML requires... in HTML5 an `<input>` tag without a `type` is assumed to hhave `type=text`.  So, as long as the behavior is the same for `element.type === undefined` as for `text`, we should be OK.  

There are other things we are not really checking such as that `l:name` fields are only used in sensible places inside forms, for example on `input` or `textarea` but not ordinary content tags.  (Of course, they should have no effect there and this patch will result in their variables being bound to `""`.)  

On balance, having dipped a toe in the chilly waters of `l:` attribute desugaring and raw forms, I lean towards thinking that modernizing and extending formlets would probably be a better use of time, to cover the use cases currently only supported by raw forms while retaining composability and nicer high-level typechecking.